examples/pub/lib/pub_repository.dart

```diff
@@ -1,15 +1,22 @@
 import 'package:dio/dio.dart';
+import 'package:flutter/foundation.dart';
 import 'package:freezed_annotation/freezed_annotation.dart';
 import 'package:pubspec_parse/pubspec_parse.dart';
 
 part 'pub_repository.freezed.dart';
 part 'pub_repository.g.dart';
 
 class PubRepository {
+  PubRepository() {
+    dio = Dio();
+    if (kIsWeb) {
+      dio.interceptors.add(PubProxyInterceptor());
+    }
+  }
+
   static const _scheme = 'https';
   static const _host = 'pub.dartlang.org';
-
-  final dio = Dio();
+  late final Dio dio;
```

Comment:
Could you remove the "late" by making a function for the `Dio()` creation?

Reply:
Got it.